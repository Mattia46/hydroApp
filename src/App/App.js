import React, { useState } from 'react';
import {
  BrowserRouter,
} from 'react-router-dom';
import { createBrowserHistory } from "history";

import {
  Routes,
  //NavigationBar,
} from '../Navigation/Navigation';
import { Header } from '../Navigation/Header/Header';
import style from '../Navigation/navigation.css';
import Mattia from '../Navigation/Navigation2';

const App = () => {
  const history = createBrowserHistory();
  const initialState = true;
  const [isAuthorized, setIsAuthorized] = useState(initialState);

  return (
    <BrowserRouter>
      <div className={style.main}>
        <Header
          history={history}
          isAuthorized={isAuthorized}
          setIsAuthorized={setIsAuthorized}
        />
        <Routes
          history={history}
          isAuthorized={isAuthorized}
          setIsAuthorized={setIsAuthorized}
        />
        <Mattia />
      </div>
    </BrowserRouter>
  );
};

export default App;

