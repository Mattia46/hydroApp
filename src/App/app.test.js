import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { getByTestId } from '@testing-library/dom';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';

import App from './App';

describe('Navigation:', () => {
    afterEach(() => {
        cleanup;
    });
    const history = createMemoryHistory();
    const getContainer = () => render(
          <Router history={history}>
            <App />
          </Router>
    );

    it('Should render the homePage with NavBar', () => {
        const { container } = getContainer();
        const navBar = getByTestId(container, "NavBar");
        expect(navBar).toBeDefined();
        expect(container).toMatchSnapshot();
    });
});

