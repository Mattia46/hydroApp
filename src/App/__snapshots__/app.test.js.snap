// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation: Should render the homePage with NavBar 1`] = `
<div>
  <div
    class="main"
  >
    <div
      class="backHeader"
    >
      <div
        class="header"
      >
        <div
          class="headerBack"
        >
          <img
            height="24"
            src="../assets/img/keyboard_backspace-24px.svg"
          />
        </div>
        <div
          class="logoHeader"
        >
          <img
            height="24"
            src="../assets/img/vinoted_logo_white.svg"
          />
        </div>
      </div>
    </div>
    <div
      class="containerBody"
    >
      <div
        data-testid="Home"
        style="height: 100%; overflow: scroll;"
      >
        <div
          class="c_homepage__header"
        >
          <div
            class="c_homepage__header_logo"
          >
            <img
              height="160"
              src="../assets/img/vinoted_logo_login.svg"
            />
          </div>
        </div>
        <div
          class="c_homepage__body"
        >
          <div>
            Vinoted
          </div>
          <div>
            Welcome to the first digital solution for professional wine taster
            <br />
             
            <br />
            <strong>
              Connected to your wine suppliers 
            </strong>
            <br />
            Meaning all the wine information are available at the tip of your fingers when attending any kind of professional wine tasting
            <br />
            <br />
            <strong>
              Digital tasting tool 
            </strong>
            <br />
            Developed to help you create precise and exhaustive tasting notes in a fast and efficient process
            <br />
            <br />
            <strong>
              Easy access 
            </strong>
            <br />
            Your tasting notes are stored in a cloud, always available on your mobile as well as computer and tablet, with filtering and search.
          </div>
        </div>
      </div>
    </div>
    <div
      class="c_bottomnav"
      data-testid="NavBar"
    >
      <a
        href="/homePage"
      >
        <div
          class="containerIconMenu"
          data-testid="homePageLinkId"
        >
          <img
            height="20"
            src="../assets/img/icon-favourites.svg"
          />
          <div
            class="iconTitle"
          >
            Home
          </div>
        </div>
      </a>
      <a
        href="/tastingCalendar"
      >
        <div
          class="containerIconMenu"
          data-testid="tastingCalendarLinkId"
        >
          <img
            height="20"
            src="../assets/img/icon-events.svg"
          />
          <div
            class="iconTitle"
          >
            Tasting Calendar
          </div>
        </div>
      </a>
      <a
        href="/favourites"
      >
        <div
          class="containerIconMenu"
          data-testid="favouritesLinkId"
        >
          <img
            height="20"
            src="../assets/img/icon-favourites.svg"
          />
          <div
            class="iconTitle"
          >
            Favourites
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;
